### PR TITLE
TVIST1-397: BBR

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/TVIST1-397

* Adds BBR helper with function to get url to “BBR-Meddelelse” for an address
* Adds BBRData entities to store BBR data fetches from the BBR api
* Extends (embedded) addresses with BBR data for display

### CODE REVIEW

File: src/Entity/ResidentComplaintBoardCase.php [hunk 1]
```diff
@@ -264,4 +264,16 @@ public function setFeePaid(bool $feePaid): self
...
         return $this;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormattedAddress(string $type): ?string
+    {
+        if ('lease' === $type) {
+            return sprintf("%s\n%s %s", $this->getLeaseStreetNameAndNumber(), $this->getLeaseZip(), $this->getLeaseCity());
```
Comment (jekuaitk) [inline src/Entity/ResidentComplaintBoardCase.php:274]: If this is used for BBR API calls city should not be necessary - im not sure if it causes issues.

File: templates/case/_case_show/_resident_and_rent_board_shareds.html.twig [hunk 2]
```diff
@@ -120,6 +120,7 @@
                             </div>
                         </div>
                         <p class="text-muted">{% trans %}Validate the legitimacy of the address by pressing the validate button{% endtrans %}</p>
+                        <a href="{{ path('case_bbr_meddelelse', {id: case.id, addressType: 'lease'}) }}">{{ 'Show BBR-Meddelelse for lease'|trans }}</a>
```
Comment (jekuaitk) [inline templates/case/_case_show/_resident_and_rent_board_shareds.html.twig:123]: Translation missing

File: src/Controller/CaseController.php [hunk 3]
```diff
@@ -188,4 +189,28 @@ public function log(CaseEntity $case): Response
...
+     *     requirements={
+     *         "_format": "pdf",
+     *     }
+     * )
+     */
+    public function bbrMeddelelse(CaseEntity $case, BBRHelper $bbrHelper, string $addressType, string $_format): Response
+    {
+        $address = $case->getFormattedAddress($addressType);
+        if (null === $address) {
+            $this->addFlash('error', 'Cannot get address');
```
Comment (jekuaitk) [inline src/Controller/CaseController.php:205]: Should be shown/rendered somehow like we discussed

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: src/Entity/ResidentComplaintBoardCase.php (hunk 1 above)
Comment (rimi-itk) [reply src/Entity/ResidentComplaintBoardCase.php:274]: For now it's only used for BBR API calls, but it makes general sense to be able to render a formatted address, e.g for future use in mail templates.

**Review by jekuaitk — APPROVED** [review]

**Review by jekuaitk — APPROVED** [review]

File: src/Service/BBRHelper.php [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+        if (null === $bbrData->getData() || $bbrData->getUpdatedAt() < new \DateTimeImmutable($this->options['ttl'])) {
+            $data = $this->fetchBBRData($address);
+            $bbrData->setData($data);
+        }
+
+        if (null !== $bbrData) {
+            $this->entityManager->persist($bbrData);
+            $this->entityManager->flush();
+        }
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply src/Service/BBRHelper.php:67]: Yes. Nice catch. I'll fix it.

**Review by jekuaitk — CHANGES_REQUESTED** [review]

File: templates/case/_bbr/data-lease.html.twig [hunk 5]
```diff
@@ -0,0 +1,2 @@
+<span style="white-space: pre">{{ data.address }}</span>
+{{ dump(data) }}
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply templates/case/_bbr/data-lease.html.twig]: No. Removed in https://github.com/itk-dev/naevnssekretariatet/pull/66/commits/489a43126dd489694f53ebad51477d7f1f653055

File: translations/messages+intl-icu.da.xlf [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" version="1.2">
+  <file source-language="en" target-language="da" datatype="plaintext" original="file.ext">
+    <header>
+      <tool tool-id="symfony" tool-name="Symfony"/>
+    </header>
+    <body>
+      <trans-unit id="fZ63rLE" resname="Close">
+        <source>Close</source>
+        <target>Luk</target>
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply translations/messages+intl-icu.da.xlf:10]: Fixed in https://github.com/itk-dev/naevnssekretariatet/pull/66/commits/dc802aa0fae142943e46d389c7ef319cfc174c42

**Review by jekuaitk — APPROVED** [review]

**Pull request merged**